### ISSUE TITLE
Improve docstring for analyze() with a full code example

### ISSUE THREAD

**Issue by Khanz9664** [Khanz9664/TrustLens#22]
# [DOCS] Improve `analyze()` docstring with a full runnable example

**Difficulty:** ⭐ Beginner  
**Estimated Time:** 15–20 minutes  

Labels: `good first issue`, `docs`, `beginner-friendly`

---

## 🔍 Problem

The current `analyze()` docstring includes only a brief example.

For new users, it’s not immediately clear how to:
- create a dataset  
- train a model  
- run `analyze()` end-to-end  

👉 We should provide a **complete, runnable example** directly in the docstring.

---

## 🎯 Task

Expand the `Examples` section of the `analyze()` function docstring.

Add a **clear, 10-line runnable example** that demonstrates:
- dataset creation using `make_classification`
- training a simple model
- running `analyze()`
- displaying results

---

## 🛠️ Where to make changes

- File: `trustlens/api.py`
- Function: `analyze()`

---

## 🧭 Getting Started

1. Open `trustlens/api.py`
2. Locate the `analyze()` function
3. Find the existing docstring
4. Navigate to the `Examples` section
5. Replace or expand it with a full working example

---

## 📌 Example (target style)

```python
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from trustlens import analyze

# Create dataset
X, y = make_classification(n_samples=500, n_features=10, random_state=42)

# Train/test split
X_train, X_test, y_train, y_test = train_test_split(X, y)

# Train model
model = RandomForestClassifier()
model.fit(X_train, y_train)

# Predict probabilities
y_prob = model.predict_proba(X_test)

# Run analysis
report = analyze(model, X_test, y_test, y_prob)

# Show results
report.show()
````

---

## ✅ Expected Outcome

* The docstring contains a **complete, copy-paste runnable example**
* Users can quickly understand how to use `analyze()`
* Improves developer experience and onboarding

---

## 🧪 How to Verify

* Open the updated docstring in your editor
* Copy the example code and run it in a Python environment
* Confirm it runs without errors and produces output

---

## 🤝 First-time contributor?

If this is your first open-source contribution, comment:

> "I'd like to work on this!"

I’ll guide you step-by-step 🚀

**Comment by q404365631** [comment]
I would like to work on this issue. Please assign it to me.

**Comment by Khanz9664** [comment]
@q404365631 Awesome, thanks for picking this up! 🚀

I've assigned the issue to you.

### Quick guidance:
- Open `trustlens/api.py`
- Locate the `analyze()` function
- Find the docstring and update the `Examples` section
- Add a clear, runnable example (like the one in the issue description)

### Tip:
Try running the example locally to make sure it works end-to-end.

Once done:
- commit your changes
- open a Pull Request

Let me know if you need any help 👍

**Comment by Khanz9664** [comment]
Hi @q404365631 🙂 

Just checking in — are you still planning to work on this?

If yes, no rush, just let me know your progress 👍  
If not, I can open it up for others as well.

Thanks!

**Comment by q404365631** [comment]
I'd like to work on this!

**Comment by Khanz9664** [comment]
Hi @q404365631 🙂

Thanks for your interest in contributing — really appreciate it!

Since multiple issues have been claimed, just a quick note to keep things moving smoothly for everyone:

👉 If you’re actively working on one of these, please share a quick progress update or open a draft PR.

👉 If there’s no activity within ~48 hours, we may unassign the issue so others can pick it up.

Also, to keep things fair:

* we usually assign **one issue at a time per contributor** (especially beginner issues)

Let me know which issue you’d like to focus on first 👍

**Comment by nguyenthekr92-byte** [comment]
I'd like to work on this bounty! I can handle documentation. I'll start working on this right away and submit a PR once done.
